crates/astria-sequencer-validation/src/utils.rs

```diff
@@ -2,17 +2,23 @@ use std::collections::BTreeMap;
 
 use crate::MerkleTree;
 
+fn concat_chain_id_and_merkle_root(chain_id: [u8; 32], root: [u8; 32]) -> [u8; 64] {
+    let mut cated = [0u8; 64];
+    cated[..32].copy_from_slice(&chain_id);
+    cated[32..].copy_from_slice(&root);
+    cated
+}
+
 /// Groups the `sequence::Action`s within the transactions by their `chain_id`.
 /// The `BTreeMap` key is the chain ID and value is the transactions.
 #[must_use]
 pub fn generate_action_tree_leaves(
-    chain_id_to_txs: BTreeMap<Vec<u8>, Vec<Vec<u8>>>,
-) -> Vec<Vec<u8>> {
-    let mut leaves: Vec<Vec<u8>> = vec![];
+    chain_id_to_txs: BTreeMap<[u8; 32], Vec<Vec<u8>>>,
```

Comment:
This makes use of the fact that chain ID is now a fixed size array. Note that `astria-proto` is not made a dependency of `astria-sequencer-validation` so the 32 bytes are not called `CHAIN_ID_LENGTH`.